### PR TITLE
Simplify `no-array-reduce` rule

### PR DESCRIPTION
**Pull request by fisker**

### CODE REVIEW

File: test/utils/not-function-types.mjs [hunk 1]
```diff
@@ -33,7 +33,7 @@ const notFunctionTypes = [
 	'~ fn',
 	'typeof fn',
 	'void fn',
-	'delete fn',
+	// 'delete fn',
```
Comment (sindresorhus) [inline test/utils/not-function-types.mjs:36]: Intentional?

Comment (fisker) [reply test/utils/not-function-types.mjs:36]: Yah, it need non-strict mode to run test, no one uses `array.reduce(delete foo)`, not worth add a seperate test.

Comment (fisker) [reply test/utils/not-function-types.mjs:37]: Added a comment

Comment (fisker) [reply test/utils/not-function-types.mjs]: I can use delete property to test :)

**Pull request merged**